### PR TITLE
Report a corrupt Bloom Pack nicely instead of crashing (BL-2485) [Bloom 3.6]

### PR DESCRIPTION
**Pull request by StephenMcConnel**
DO NOT MERGE UNTIL 3.5 HAS ITS OWN BRANCH!

[Review on Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/881)

### CODE REVIEW

**Comment by JohnThomson** [comment]
LGTM

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/881#-:-K3Rb98n1bX_AA_mq5i6:676834233)_

File: src/BloomExe/Collection/BloomPack/BloomPackInstallDialog.cs [hunk 1]
```diff
@@ -167,6 +187,12 @@ private void _backgroundWorker_DoWork(object sender, System.ComponentModel.DoWor
 					}
 				}
 			}
+			catch (Exception ex)
+			{
+				// Report a corrupt file instead of crashing.  See http://issues.bloomlibrary.org/youtrack/issue/BL-2485.
+				ReportErrorUnzippingBloomPack();
```
Comment (hatton) [inline src/BloomExe/Collection/BloomPack/BloomPackInstallDialog.cs]: Is it safe for a background worker to touch UI controls without an Invoke?

Comment (JohnThomson) [inline src/BloomExe/Collection/BloomPack/BloomPackInstallDialog.cs]: Pretty sure it's not. Is this reporting happening on a background thread?

**Comment by StephenMcConnel** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

_[src/BloomExe/Collection/BloomPack/BloomPackInstallDialog.cs, line 193 \[r2\]](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/881#-K3RQkdNDf-uJOAU6WSI-r2-193:-K3WtXe9qELJaIdW8C88:-548983231) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0ab6623401a14f102d1c4e77fb6e6ffd417725a5/src/BloomExe/Collection/BloomPack/BloomPackInstallDialog.cs#L193)):_
I've added code to use Invoke if needed for all three places that report an error.

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/881)_

**Comment by JohnThomson** [comment]
LGTM

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/881#-:-K3afOusmRuANBsfzQnX:676834233)_

**Pull request merged**